src/yz_solrq_helper.erl

```diff
@@ -0,0 +1,225 @@
+    gen_server:cast(HPid, {batch, Index, Entries}).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+init([]) ->
+    %% TODO: Prolly need to monitor
+    %% Host = "localhost",
+    %% Port = yz_solr:port(),
+    %% {ok, HttpPid} = ibrowse_http_client:start({Host, Port}),
+    {ok, #state{}}.
+
+handle_call(status, _From, #state{qpid = QPid,
+                                  http_pid = HttpPid}) ->
+    {reply, [{qpid, QPid},
+             {http_pid, HttpPid}]};
+handle_call(BadMsg, _From, State) ->
+    {reply, {error, {unknown, BadMsg}}, State}.
+
+handle_cast({ready, Index, QPid}, State) ->
+    yz_solrq:request_batch(QPid, Index, self()),
+    {noreply, State};
+handle_cast({batch, Index, Entries0}, State) ->
+    try
+        %% TODO: use ibrowse http worker
+        %% TODO: batch updates to YZ AAE
+        Entries = [{BKey, Obj, Reason, P, riak_kv_util:get_index_n(BKey),
+                    yz_kv:hash_object(Obj)} ||
+                      {BKey, Obj, Reason, P} <- Entries0],
+        case update_solr(Index, Entries) of
+            ok ->
+                update_aae_and_repair_stats(Entries);
+            {error, Reason} ->
+                ok
+        end,
+        {noreply, State}
+    catch
+        _:Err ->
+            yz_stat:index_fail(),
+            Trace = erlang:get_stacktrace(),
+            ?ERROR("index failed - ~p\nat: ~p", [Err, Trace]),
+            {noreply, State}
+    end.
+
+%% Entries is [{Index, BKey, Obj, Reason, P, ShortPL, Hash}]
+update_solr(Index, Entries) ->
+    case yz_kv:should_index(Index) of
+        false ->
+            ok; % No need to send anything to SOLR, still need for AAE.
+        _ ->
+            IndexName = (?BIN_TO_ATOM(Index)),
+            case yz_fuse:check(IndexName) of
+                ok ->
+                    send_solr_ops(Index, solr_ops(Entries));
+                blown ->
+                    ?ERROR("Fuse Blown: can't current send solr "
+                           "operations for index ~s", [Index]),
+                    {error, fuse_blown};
+                _ ->
+                    %% fuse table creation is idempotent and occurs on
+                    %% add_index/1 on 1st creation or diff-check.
+                    %% We send entries until we can ask again for
+                    %% ok | error, as we wait for the tick.
+                    send_solr_ops(Index, solr_ops(Entries))
+            end
+    end.
+
+%% Build the SOLR query
+solr_ops(Entries) ->
+    Ring = yz_misc:get_ring(transformed),
+    LI = yz_cover:logical_index(Ring),
+    lists:reverse(
+      lists:foldl(
+        fun({BKey, Obj0, Reason0, P, ShortPL, Hash}, Ops) ->
+            %% TODO: This should be called in yz_solr:index
+            %% then the ring lookup can be removed
+            case yz_kv:is_owner_or_future_owner(P, node(), Ring) of
```

Comment:
Helper will add AAE entries for fallbacks now. Previous behavior was only for primaries.  Fixed.